oneapp-server/dao/LocationDao.js

```diff
@@ -0,0 +1,22 @@
+const { SQLDataSource } = require('datasource-sql');
+
+class LocationDao extends SQLDataSource {
+  async getCounties() {
+    const query = this.knex.column('COUNTY_NUMBER', 'COUNTY_NAME')
+      .select()
+      .from('OA_COUNTY');
+    return query.cache(60 * 60);
+  }
+
+  async isValidNJZipcode(ZIPCODE, COUNTY_NUMBER) {
+    const zipcodeCount = await this.knex
+      .from('OA_COUNTY_ZIP_LOOKUP')
+      .where({ ZIP_CODE: ZIPCODE })
+      .where({ COUNTY_NUMBER })
+      .count('ZIP_CODE as zipcodes')
+      .first();
```

Comment:
I reviewed the existing stored procedure `SP_VALIDATE_ZIPCODE` and the existing stored procedure also takes into account the `IS_HOMELESS` flag: https://github.com/newjersey/dhs-oneapp-legacy/blob/master/OneAppStoredProceduresFunctions/OA_PKG_APP.sql#L813

While we could write a separate assertion to ignore the zipcode validation if the applicant is homeless, I think it makes the most sense to mirror the existing functionality, as we'll call this not only on the front-end, but in the update contact logic. Could you swap this out for the stored procedure call `SP_VALIDATE_ZIPCODE` and add in an extra required parameter, `IS_HOMELESS`?